compiler/luci/partition/src/CircleNodeClone.cpp

```diff
@@ -0,0 +1,106 @@
+/*
+ * Copyright (c) 2021 Samsung Electronics Co., Ltd. All Rights Reserved
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "CircleNodeClone.h"
+
+#include <oops/UserExn.h>
+
+namespace
+{
+
+class CloneNode final : public luci::CircleNodeVisitor<luci::CircleNode *>
+{
+public:
+  CloneNode(loco::Graph *graph) : _graph(graph){};
+
+public:
+  luci::CircleNode *visit(const luci::CircleAdd *) final;
+  // TODO add all nodes
+
+  // NOTE CircleNodeVisitor will throw if not supported here
+
+protected:
+  loco::Graph *_graph = nullptr;
+};
+
+luci::CircleNode *CloneNode::visit(const luci::CircleAdd *node)
+{
+  auto *cloned = _graph->nodes()->create<luci::CircleAdd>();
+  cloned->fusedActivationFunction(node->fusedActivationFunction());
+  return cloned;
+}
+
+} // namespace
+
+namespace luci
+{
+
+void copy_common_attributes(const luci::CircleNode *src, luci::CircleNode *dst)
+{
+  assert(src != nullptr);
+  assert(dst != nullptr);
+
+  dst->name(src->name());
+  dst->dtype(src->dtype());
+
+  dst->rank(src->rank());
+  for (uint32_t i = 0; i < src->rank(); i++)
+  {
+    if (src->dim(i).known())
+      dst->dim(i).set(src->dim(i).value());
```

Comment:
```suggestion
    dst->dim(i) = src->dim(i);
```

---

Following codes should be fixed too but we may handle it while relocating/refactoring `CircleNodeClone.h`...
https://github.com/Samsung/ONE/blob/5c4d2f72a3837c4f3453746ac0ea4a87479a50e6/compiler/luci/service/src/Nodes/CircleConst.cpp#L56-L57